docker

```diff
@@ -0,0 +1,192 @@
+    echo "  cli     - installs plugin into OMERO.server"
+    echo "  lib     - builds and runs client-side library"
+    echo "  scripts - installs OMERO.scripts into OMERO.server"
+    echo
+    echo "Environment variables:"
+    echo "  PROJECT - name of the docker-compose project ($PROJECT)"
+    echo "  NETWORK - name of the docker network to use ($NETWORK)"
+    echo "  TARGET  - name of the component being installed ($TARGET)"
+    echo "  ACTION  - use 'echo' to perform a dry-run ($ACTION)"
+    echo "  TRAVIS  - whether to print travis_folds ($TRAVIS)"
+    echo "  VERBOSE - set to 'set -x' to see all actions ($VERBOSE)"
+    exit 2
+else
+    STAGES="$@"
+fi
+
+$VERBOSE
+
+##
+## Start-up containers so we can configure them
+##
+$ACTION .omero/compose up -d --force-recreate
+
+##
+## Stages
+##
+
+fold() {
+    if [ "$TRAVIS" == "true" ]; then
+        printf "travis_fold:$1:$2\n"
+    fi
+}
+
+run() {
+    # Primary task which uses the first two arguments to either:
+    #  1) run .omeroci/$1-$2 if it exists, or
+    #  2) run the default script in this directory of the same name.
+
+    LOCAL=false
+    CALLAS="-u root"
+    if [ $1 == "--local" ]; then
+        LOCAL=true
+        shift
+    elif [ $1 == "--user" ]; then
+        CALLAS="-u $USER"
+        shift
+    fi
+
+    type=$1
+    name=$2
+    file="$1-$2"
+    var="$(echo $1| tr /a-z/ /A-Z/)_$(echo $2 | tr /a-z/ /A-Z/)"
+    eval "$var=\${$var:-0}"
+
+    if [ "${!var}" == "0" ]; then
+        eval "$var=1"
+        fold start $file
+        if [ -e .omeroci/$file ]; then
+            PREFIX=/$TARGET
+            LOCALPREFIX=""
+            ARGUMENT=.omeroci/$file
+        elif [ -e $(dirname "$0")/$file ]; then
+            PREFIX=/infra
+            LOCALPREFIX=".omero"
+            ARGUMENT=$file
+        elif [ -e $(dirname "$0")/$STAGE-$name ]; then
+            PREFIX=/infra
+            LOCALPREFIX=".omero"
+            ARGUMENT=$STAGE-$name
+        else
+            echo "No script found:"
+            echo "  .omeroci/$file"
+            echo "  $(dirname "$0")/$file"
+            echo "  $(dirname "$0")/$STAGE-$name"
+            return 2
+        fi
+        if [ "$LOCAL" == true ]; then
+            $ACTION $LOCALPREFIX/$ARGUMENT
+        else
+            $ACTION docker exec $CALLAS -e TARGET=$TARGET $CID $PREFIX/$ARGUMENT
+        fi
+        fold end $file
+    fi
+}
+
+install() {
+    $ACTION docker cp -L .      $CID:/$TARGET
+    $ACTION docker cp -L .omero $CID:/infra
+    $ACTION docker exec -u root -e TARGET=$TARGET $CID sh -c "chown -R $USER:$USER $TARGET /infra"
+}
+
+java() {
+    $ACTION docker exec -e USER_AGENT=Travis -u root $CID /opt/omero/omego/bin/omego download java --sym /opt/omero/java
+    $ACTION docker exec $CID ln -s /opt/omero/java/libs /opt/omero/web/OMERO.web/lib/client
+}
+
+wait_on_login() {
+    $ACTION docker exec -e OMERO_HOST=omero $CID /infra/wait-on-login
+}
+
+
+app_config() {
+    # TODO: add to run
```

Comment:
If NAME detection is done in utils, then this method can be removed in favor of `run app config`